sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/FilterEstimation.scala

```diff
@@ -101,21 +101,23 @@ case class FilterEstimation(plan: Filter, catalystConf: CatalystConf) extends Lo
         }
 
       case Or(cond1, cond2) =>
-        // For ease of debugging, we compute percent1 and percent2 in 2 statements.
         val percent1 = calculateFilterSelectivity(cond1, update = false)
         val percent2 = calculateFilterSelectivity(cond2, update = false)
         (percent1, percent2) match {
           case (Some(p1), Some(p2)) => Some(math.min(1.0, p1 + p2 - (p1 * p2)))
-          case (Some(p1), None) => Some(1.0)
-          case (None, Some(p2)) => Some(1.0)
-          case (None, None) => None
+          case _ => None
         }
 
-      case Not(cond) => calculateFilterSelectivity(cond, update = false) match {
-        case Some(percent) => Some(1.0 - percent)
-        // for not-supported condition, set filter selectivity to a conservative estimate 100%
-        case None => None
-      }
+      case Not(cond) =>
+        if (cond.isInstanceOf[And] || cond.isInstanceOf[Or]) {
+          // Don't support compound Not expression.
```

Comment:
our goal is to not under-estimate, so `((NOT cond1) && (NOT cond2))` should be allowed, `NOT(cond1 && cond2)` should not be, because we don't know if `cond1` or `cond1` is over-estimated or not.